tests/unit/test_numpy_splitters.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+import pandas as pd
+import numpy as np
+import pytest
+
+from reco_utils.dataset.numpy_splitters import numpy_stratified_split
+
+
+@pytest.fixture(scope="module")
+def test_specs():
+
+    return {
+        "number_of_items": 50,
+        "number_of_users": 20,
+        "seed": 123,
+        "ratio": 0.6,
+        "tolerance": 0.01,
+        "fluctuation": 0.02,
+    }
+
+
+@pytest.fixture(scope="module")
+def python_dataset(test_specs):
+
+    """Generate a test user/item affinity Matrix"""
+
+    # fix the the random seed
+    np.random.seed(test_specs["seed"])
+
+    # generates the user/item affinity matrix. Ratings are from 1 to 5, with 0s denoting unrated items
+    X = np.random.randint(
+        low=0,
+        high=6,
+        size=(test_specs["number_of_users"], test_specs["number_of_items"]),
+    )
+
+    return X
+
+
+def test_random_stratified_splitter(test_specs, python_dataset):
+    """
+    Test the random stratified splitter.
+    """
+
+    # generate a syntetic dataset
+    X = python_dataset
+
+    # the splitter returns (in order): train and test user/affinity matrices, train and test datafarmes and user/items to matrix maps
+    Xtr, Xtst = numpy_stratified_split(
+        X, ratio=test_specs["ratio"], seed=test_specs["seed"]
+    )
+
+    # Tests
+    # check that the generated matrices have the correct dimensions
+    assert (Xtr.shape[0] == X.shape[0]) & (Xtr.shape[1] == X.shape[1])
+
+    assert (Xtst.shape[0] == X.shape[0]) & (Xtst.shape[1] == X.shape[1])
+
+    X_rated = np.sum(X != 0, axis=1)  # number of total rated items per user
+    Xtr_rated = np.sum(Xtr != 0, axis=1)  # number of rated items in the train set
+    Xtst_rated = np.sum(Xtst != 0, axis=1)  # number of rated items in the test set
+
+    # global split: check that the all dataset is split in the correct ratio
+    assert (
```

Comment:
Couldn't you use `pytest.approx()` for these as in other tests?

Reply:
Hi Andreas, I have added back the `pytest.approx()` method for the test of the global split. On the other hand, this method does not work with the "local" test, where the assert is over an array. This was the original reason for not using the `pytest.approx()` method. Nevertheless, there may be a simple way to make `pytest.approx()` works for an array, please let me know if you have any suggestions.

Reply:
For scalar values we use `pytest.approx()` a lot. For arrays, I think it should work too 
https://docs.pytest.org/en/latest/reference.html#pytest-approx
but I don't recall if we have used it.

Reply:
Thanks for the link, I found the problem and now it works with `pytest.approx()` both for scalars and arrays :)